internal/fwserver/server_planresourcechange.go

```diff
@@ -196,9 +217,10 @@ func (s *Server) PlanResourceChange(ctx context.Context, req *PlanResourceChange
 		logging.FrameworkTrace(ctx, "Resource implements ResourceWithModifyPlan")
 
 		modifyPlanReq := resource.ModifyPlanRequest{
-			Config: *req.Config,
-			Plan:   stateToPlan(*resp.PlannedState),
-			State:  *req.PriorState,
+			Config:  *req.Config,
+			Plan:    stateToPlan(*resp.PlannedState),
+			State:   *req.PriorState,
+			Private: privateProviderData,
```

Comment:
Similarly here regarding using the updated private state data from the schema handling above -- this can be unit tested by implementing both schema and resource level modifications where the resource performs a conditional update based on the result of a schema change.